superset/common/query_object.py

```diff
@@ -336,6 +335,7 @@ def to_dict(self) -> Dict[str, Any]:
             "series_limit": self.series_limit,
             "series_limit_metric": self.series_limit_metric,
             "to_dttm": self.to_dttm,
+            "time_shift": self.time_shift,
```

Comment:
I believe this will invalidate all currently stored cache values: to avoid cache misses, I've previously placed a check on `QueryObject.cache_key` checking if the new property is truthy, and if so, add a new key to the cache key dict (the idea is to move all of these in one go to `QueryObject.to_dict()` to make a big bang during e.g. a major release).

Reply:
`time_shift` is not really useful in existing code. It seems to be a parameter used by the old time picker, and I'm only using it here to ensure backward compatibility for the query object.